src/celldega/qc/__init__.py

```diff
@@ -0,0 +1,308 @@
+                                 path_transformation_matrix=transform_file,
+                                 )
+        
+        trx = trx.to_pandas()
+        trx = trx.rename(columns={'transformed_x': 'global_x', 'transformed_y': 'global_y', 'name': gene})
+        trx_meta = trx[trx[cell_index] != -1][[transcript_index, cell_index, gene]]
+
+        transformation_matrix = pd.read_csv(transform_file, header=None, sep=" ").values
+
+        cell_gdf = get_cell_polygons(technology=technology_name, 
+                                     path_cell_boundaries=cell_boundaries_file, 
+                                     transformation_matrix=transformation_matrix,
+                                     path_output=path_output,
+                                     path_meta_cell_micron=path_meta_cell_micron)
+        
+        cell_gdf.reset_index(inplace=True)
+        cell_gdf['area'] = cell_gdf['geometry'].area
+        cell_gdf['centroid'] = cell_gdf['geometry'].centroid
+        cell_meta_gdf = cell_gdf[['cell_id', 'area', 'centroid']]
+
+    percentage_of_assigned_transcripts = (len(trx_meta) / len(trx))
+
+    metrics['dataset_name'] = dataset_name
+    metrics['segmentation_approach'] = segmentation_approach
+    
+    metrics['proportion_assigned_transcripts'] = percentage_of_assigned_transcripts
+    metrics['number_cells'] = len(cell_gdf)
+    metrics['mean_cell_area'] = cell_gdf['geometry'].area.mean()
+    
+    metrics['mean_transcripts_per_cell'] = trx_meta.groupby(cell_index).size().mean()
+    metrics['median_transcripts_per_cell'] = trx_meta.groupby(cell_index)[transcript_index].count().median()
+
+    metrics['average_genes_per_cell'] = trx_meta.groupby(cell_index)[gene].nunique().mean()
+    metrics['median_genes_per_cell'] = trx_meta.groupby(cell_index)[gene].nunique().median()
+
+    metrics['proportion_empty_cells'] = ((len(cell_meta_gdf) - len(cell_gdf)) / len(cell_meta_gdf))
+
+    metrics_df = pd.DataFrame([metrics])
+    metrics_df = metrics_df.T
+    metrics_df.columns = [f"{dataset_name}-{segmentation_approach}"]
+    metrics_df = metrics_df.T
+    
+    gene_specific_metrics_df = pd.DataFrame({
+        "proportion_of_cells_expressing": (trx_meta.groupby(gene)[cell_index].nunique()) / len(cell_gdf),
+        "average_expression": (trx_meta.groupby(gene)[cell_index].nunique()) / (trx_meta.groupby(gene)[cell_index].nunique().sum()),
+        "assigned_transcripts": (trx_meta.groupby(gene)[transcript_index].count() / trx.groupby(gene)[transcript_index].count()).fillna(0)
+    })
+
+    metrics_df.to_csv(path_output_cell_metrics)
+    gene_specific_metrics_df.to_csv(path_output_gene_metrics)
+
+    print("segmentation metrics calculation completed")
+
+def mixed_expression_calc(default_segmentation_segmentation_name, default_segmentation_cell_feature_matrix_path, algorithm_names, algorithm_specific_cbg_files, cell_type_A_specific_genes, cell_type_B_specific_genes, cell_A_name, cell_B_name):
```

Comment:
Can you add `cmap` as an argument, just in case people want to use a different color map. You can assign a default value though so it is backward compatible.